### ISSUE TITLE
wav2lip_train training error：Expected more than 1 value per channel when training, got input size torch.Size([1, 512, 1, 1])

### ISSUE THREAD

**Issue by cncbec** [Rudrabha/Wav2Lip#563]
![image](https://github.com/Rudrabha/Wav2Lip/assets/54312958/b6133394-0868-4348-b4be-70828f305de7)
Using my personal data set, the batch size in hparams.py is 16

**Comment by ulucsahin** [comment]
Have you been able to resolve this issue? I am also using a custom dataset and getting the same error.

**Comment by cncbec** [comment]
> Have you been able to resolve this issue? I am also using a custom dataset and getting the same error.

No,I haven't

**Comment by nicoleng** [comment]
I just resolved this issue by adding syncnet.eval() after 
`syncnet = SyncNet().to(device)

@cncbec @ulucsahin I am also using a custom dataset and my guess is either this error was not encountered with the original dataset or the model was being set to validate in the eval functions included for LRS2 data which I'm not using

**Comment by Ezrealz** [comment]
I experienced the same problem, and whenever sync loss is reduced to 0.75 or less, an error is reported

**Comment by lishiqi777** [comment]
上万高清中文数据集训练出来的256高清模型，含推理代码，oonx加速推理。
需要+V  tke365  先看效果再做决定

**Comment by bjfrbjx** [comment]
!!! 因为pytorch在训练时会使用BatchNormal，就要保证batch_size>1，所以在model.train()下会触发_verify_batch_size()判断。
最后解决只需要准备两个以上的样例

**Comment by Jayzeen13** [comment]
> I just resolved this issue by adding syncnet.eval() after `syncnet = SyncNet().to(device)
> 
> @cncbec @ulucsahin I am also using a custom dataset and my guess is either this error was not encountered with the original dataset or the model was being set to validate in the eval functions included for LRS2 data which I'm not using

Hi, Sorry for bothering you, but since you are using a custom dataset, can you please show me the structure of the filelist for the custom dataset. because when i try to train on my own dataset, the progress is 0% always and i think it is an issue with the filelist(Ex: train.txt file)

**Comment by Rykertang** [comment]
> @nicoleng
> 
> > I just resolved this issue by adding syncnet.eval() after `syncnet = SyncNet().to(device)我刚刚通过在 `syncnet = SyncNet().to(device) 之后添加syncnet.eval() 解决了这个问题
> > @cncbec @ulucsahin I am also using a custom dataset and my guess is either this error was not encountered with the original dataset or the model was being set to validate in the eval functions included for LRS2 data which I'm not using我也在使用自定义数据集，我的猜测是原始数据集没有遇到此错误，或者模型被设置为在我没有使用的 LRS2 数据包含的 eval 函数中进行验证
> 
> Hi, Sorry for bothering you, but since you are using a custom dataset, can you please show me the structure of the filelist for the custom dataset. because when i try to train on my own dataset, the progress is 0% always and i think it is an issue with the filelist(Ex: train.txt file)您好，很抱歉打扰您，但由于您使用的是自定义数据集，您能否向我展示自定义数据集的文件列表的结构。因为当我尝试在自己的数据集上训练时，进度始终为 0%，我认为这是文件列表的问题（例如：train.txt 文件）

Hello, may I ask if this issue has been resolved? I have also encountered this problem

**Comment by Rykertang** [comment]
> !!! 因为pytorch在训练时会使用BatchNormal，就要保证batch_size>1，所以在model.train()下会触发_verify_batch_size()判断。 最后解决只需要准备两个以上的样例

这个是什么意思，我现在batchsize已经设置为16了还是报错这个
